tools/src/tester/ChapterEvaluator.ts

```diff
@@ -10,40 +10,43 @@
 import { type Chapter, type ActualResponse } from './types/story.types'
 import { type ChapterEvaluation, type Evaluation, Result } from './types/eval.types'
 import { type ParsedOperation } from './types/spec.types'
-import { overall_result } from './helpers'
+import { extract_output_values, overall_result } from './helpers'
 import type ChapterReader from './ChapterReader'
 import type OperationLocator from './OperationLocator'
 import type SchemaValidator from './SchemaValidator'
+import { type StoryOutputs } from './StoryOutputs'
 
 export default class ChapterEvaluator {
   private readonly _operation_locator: OperationLocator
   private readonly _chapter_reader: ChapterReader
   private readonly _schema_validator: SchemaValidator
 
-  constructor (spec_parser: OperationLocator, chapter_reader: ChapterReader, schema_validator: SchemaValidator) {
+  constructor(spec_parser: OperationLocator, chapter_reader: ChapterReader, schema_validator: SchemaValidator) {
     this._operation_locator = spec_parser
     this._chapter_reader = chapter_reader
     this._schema_validator = schema_validator
   }
 
-  async evaluate (chapter: Chapter, skip: boolean): Promise<ChapterEvaluation> {
+  async evaluate(chapter: Chapter, skip: boolean, story_outputs: StoryOutputs): Promise<ChapterEvaluation> {
     if (skip) return { title: chapter.synopsis, overall: { result: Result.SKIPPED } }
-    const response = await this._chapter_reader.read(chapter)
+    const response = await this._chapter_reader.read(chapter, story_outputs)
     const operation = this._operation_locator.locate_operation(chapter)
     if (operation == null) return { title: chapter.synopsis, overall: { result: Result.FAILED, message: `Operation "${chapter.method.toUpperCase()} ${chapter.path}" not found in the spec.` } }
     const params = this.#evaluate_parameters(chapter, operation)
     const request_body = this.#evaluate_request_body(chapter, operation)
     const status = this.#evaluate_status(chapter, response)
     const payload = status.result === Result.PASSED ? this.#evaluate_payload(response, operation) : { result: Result.SKIPPED }
+    const output_values = extract_output_values(response, chapter.output)
```

Comment:
`extract_output_values` should either be a private method on `ChapterEvaluator` or maybe this works better as `chapter.output.extract_output_values_from_response(response)`?